shared-bindings/storage/__init__.c

```diff
@@ -150,7 +150,7 @@ STATIC mp_obj_t storage_getmount(const mp_obj_t mnt_in) {
 }
 MP_DEFINE_CONST_FUN_OBJ_1(storage_getmount_obj, storage_getmount);
 
-//| def erase_filesystem() -> None:
+//| def erase_filesystem(extended: Optional[bool] = True) -> None:
```

Comment:
I think we need to discuss what the default value of this and dualbank.switch(both_bootable=) should be. Also, what the action in the case of corrupt CIRCUITPY is, and whether it should match the default of an explicit `erase_filesystem()` with its default value. If this has already been discussed, a pointer to the discussion would be helpful.

Reply:
I chose to make storage extended by default as I think more people would want an extended storage over ota updates.

Reply:
Now that you pointed it out, I realized that this leaves the possibility where the user wants both partitions to contain bootable image but corrupts the CIRCUITPY, calls just `erase_filesystem()` and ends up having the other image wiped.

Reply:
Using the following API now:
```py
def erase_filesystem(extended: Optional[bool] = None) -> None:
    """Erase and re-create the ``CIRCUITPY`` filesystem.

    :param bool extended: On boards that support ``CIRCUITPY_DUALBANK``
        and ``CIRCUITPY_STORAGE_EXTEND``, the ``CIRCUITPY`` storage can be
        extended by setting this to `True`. If this isn't provided or set to
        `None` (default), existing configuration will be used."""
```
Using `dualbank` when storage is extended will raise an error.
```c
STATIC void raise_error_if_storage_extended(void) {
    if (supervisor_flash_get_extended()) {
        mp_raise_RuntimeError(translate("storage is extended"));
    }
}
```